terra/executor/thread.py

```diff
@@ -1,9 +1,39 @@
 import concurrent.futures
+
 import terra.executor.base
+import terra.core.settings
 
 __all__ = ['ThreadPoolExecutor']
 
 
 class ThreadPoolExecutor(concurrent.futures.ThreadPoolExecutor,
                          terra.executor.base.BaseExecutor):
-  pass
+  '''
+  Terra version of :class:`concurrent.futures.ThreadPoolExecutor`
+
+  Unlike other executors, :class:`ThreadPoolExecutor` has no process isolation.
+  This results in a scenario where multiple threads could be terra settings,
```

Comment:
This results in a scenario where multiple threads using terra settings could actually influence

Reply:
That was supposed to read `This results in a scenario where multiple threads could use terra settings to influence each other`